services/py-api/src/database/db_manager.py

```diff
@@ -0,0 +1,88 @@
+import os
+from typing import Annotated
+
+from fastapi import Depends
+from motor.motor_asyncio import AsyncIOMotorClient
+from pymongo import MongoClient
+from pymongo.errors import ConnectionFailure, OperationFailure, ConfigurationError
+from result import Err
+from structlog.stdlib import get_logger
+
+from src.utils import SingletonMeta
+
+LOG = get_logger()
+
+
+class DatabaseManager(metaclass=SingletonMeta):
+    """Creates a Thread-safe singleton Database manager. It provides a wrapper over the AsyncIOMotorClient and utils for
+    pinging the database. The singleton behaviour allows us to have one instance of the AsyncIOMotorClient which we
+    could safely use across our application through the interface of the DatabaseManager"""
+
+    DB_NAME = "TheHubDEV"
+
+    def __init__(self) -> None:
+        # The mongo client has a conn pool under the hood, and we set a min number of idle conn that the pool has
+        # to maintain, the default is 0. After maxIdleTimeMS the connection pool replaces the idle conn with a new one.
+        # I set a min number of conn according to the article below. The connection is also removed too fast, after
+        # around 30 sec, that's why I increased maxIdleTimeMS to 5 min.
+        # https://alexedwards.net/blog/configuring-sqldb
+        # https://www.mongodb.com/docs/languages/python/pymongo-driver/current/faq/#how-does-connection-pooling-work-in-pymongo-
+        self._client = AsyncIOMotorClient(host=os.environ["DATABASE_URL"], minPoolSize=25, maxIdleTimeMS=5 * 60 * 1000)
+
+    def close_all_connections(self) -> Err[str]:
+        """Closes all connections in the pool managed by Mongo"""
+        if not self._client:
+            return Err("The database client is not initialized!")
+
+        LOG.debug("Closing all database connections")
+        self._client.close()
+
+    async def async_ping_db(self) -> Err[str]:
+        try:
+            LOG.debug("Pinging MongoDB...")
+            await self._client.admin.command("ping")
+        except ConnectionFailure:
+            return Err("Database is unavailable!")
+        except (OperationFailure, ConfigurationError):
+            return Err("Database authentication failed!")
+
+        LOG.debug("Pong")
+
+    @property
+    def client(self) -> AsyncIOMotorClient:
+        return self._client
+
+
+def ping_db() -> Err[str]:
+    """This method is used on application startup. It is not async as I cannot await the start() method"""
```

Comment:
This comment is confusing. Why do you need an async method? 

It's a bit strange you're adding a docstring comment and talking in first person. If this is not a comment which is to be exposed by the lsp utilities, have it as a regular comment.

Reply:
Fixed in [17e3ee7](https://github.com/AUBGTheHUB/monolith/pull/716/commits/17e3ee70284456f6830134d48e3e1855c1afd301)